File: src/react/maze_web_server/src/utils/dailyChallenge.ts
import { getFeaturedGameItems, getGameCollection } from '../api/client'
import { launchDefinition } from './play3dLaunch'

// The curated collection the daily games live in (seeded at server startup); the
// Today's Challenge entry points find it in the featured catalogue by this name.
const DAILY_CHALLENGES_COLLECTION = 'Daily Challenges'

// Resolve and launch today's daily challenge — client-side, no dedicated
// endpoint: find the curated "Daily Challenges" collection in the featured
// catalogue, resolve its detail, and launch its daily member (the host page
// date-mixes the seed for the current UTC day). Returns true when a game was
// launched, false when there's nothing to play (no collection or no member);
// throws on a network failure so the caller can distinguish "unavailable" from
// "load failed". Shared by the Home tile and the hamburger menu.
export async function launchTodaysChallenge(token: string): Promise<boolean> {
  const featured = await getFeaturedGameItems(token, { limit: 100 })
  const collection = featured.items.find(
    i => i.kind === 'collection' && i.collection?.name === DAILY_CHALLENGES_COLLECTION,
  )?.collection
  const detail = collection ? await getGameCollection(token, collection.id) : null
  const daily = detail?.definitions.find(d => d.rotation === 'daily') ?? detail?.definitions[0]
  if (!daily) return false
  launchDefinition(daily.id)
  return true
}

File: src/csharp/Maze.Maui.App/Services/DailyChallenge.cs
using Maze.Maui.App.Models;

namespace Maze.Maui.App.Services
{
    /// <summary>
    /// Resolves today's daily challenge from the featured catalogue — the C#
    /// mirror of the web client's <c>launchTodaysChallenge</c>. The daily games
    /// live in a curated "Daily Challenges" collection (seeded at server startup),
    /// so the Today's Challenge entry point finds it by name and plays its daily
    /// member (the host page date-mixes the seed for the current UTC day). Pure, so
    /// the resolution is unit-testable without navigation or HTTP.
    /// </summary>
    public static class DailyChallenge
    {
        /// <summary>The curated collection the daily games live in.</summary>
        public const string CollectionName = "Daily Challenges";

        /// <summary>
        /// Finds the "Daily Challenges" collection in the featured catalogue.
        /// </summary>
        /// <param name="items">The featured catalogue items</param>
        /// <returns>The collection, or <c>null</c> when it isn't featured</returns>
        public static GameCollection? FindCollection(IReadOnlyList<FeaturedGameItem> items)
        {
            foreach (FeaturedGameItem item in items)
            {
                if (item.Collection is not null && item.Collection.Name == CollectionName)
                    return item.Collection;
            }
            return null;
        }

        /// <summary>
        /// Picks the member to play — the first <c>daily</c>-rotation game, else
        /// the first member (a defensive fallback).
        /// </summary>
        /// <param name="members">The collection's member definitions</param>
        /// <returns>The game to play, or <c>null</c> when the collection is empty</returns>
        public static GameDefinition? PickDaily(IReadOnlyList<GameDefinition> members)
        {
            foreach (GameDefinition member in members)
            {
                if (member.Rotation == GameVocabulary.Rotation.Daily)
                    return member;
            }
            return members.Count > 0 ? members[0] : null;
        }
    }
}
